### DISCUSSION TITLE
Library Source - CDE (Various)

### DISCUSSION THREAD

**Discussion by GeoffLyle** [CBIIT/ccdi-federation-api#46] [Metadata Field Ratification]
caDSR Link: [6285979](https://cadsr.cancer.gov/onedata/dmdirect/NIH/NCI/CO/CDEDD?filter=CDEDD.ITEM_ID=6285979%20and%20ver_nr=1)

Definition: "The source of a sample collection of double stranded DNA fragments analyzed by high-throughput sequencing."

Issues with CDE:
The permissible values in this CDE encompass two orthogonal concepts, source material and source molecule.

Proposed short term solution as of December 8, 2023:
Subset the CDE:

- Library Source Material (Bulk cells, Bulk nuclei, Bulk tissue, Single-cells, Single-Nuclei)​

- Library Source Molecule (Genomic DNA, Genomic Single Cell, Metagenomic, Metatranscriptomic, Synthetic, Transcriptomic, Transcriptomic Single Cell, Viral Small RNA)

Note: Clay mentioned 'Genomic Single Cell' and 'Transcriptomic Single Cell' confound the two concepts and should be not be included in either subset.

**Comment by GeoffLyle** [comment]
Slack Discussion:

_From Clay McLeod on October 25th, 2023_ - I again find the permissible values are not at the same level of granularity and are therefore orthogonal. For example, some permissible values address the source molecule (e.g., "Genomic DNA", "RNA", "Transcriptomic"), others specify the cellular context ("Bulk cells", "Single-cells"), and some are combinations of the two ("Genomic Single Cell"). It is difficult to know how to annotate these with a single value. If I have a single-cell sample that sequences the genomic DNA, is that marked as "Single-cells", "Genomic DNA", "Genomic Single Cell", or some combination of the three? Another example would be whole genome sequencing done on a population of bulk cells: is that "Genomic DNA", "Bulk cells", both? Perhaps even it is "Bulk Tissue", as the implication within that description is that there is a heterogenous population of cells (though the "bulk cells" value makes no mention of a homogenous population of cells to distinguish itself).

_From Patrick Dunn on November 27th, 2023_ - If the two different harmonized fields and their subsetted values are available, FNL can pass this along to cDSR for their review and implementation.

_Geoff Lyle's data harmonization subgroup meeting notes on November 17th, 2023_ -  
Proposed Solutions:

- Subset the permissible values by the orthogonal concepts.  Create two different harmonized fields based on the subsetted values from the one CDE.
- Create two new CDEs based on these subsets of the current permissible values in the Library Source CDE.

**Comment by jknable** [comment]
We can separate this CDE into new CDEs to accommodate the two intents - **Source Material** and **Source Molecule**.  We would associate the response values as listed in this GitHub post. 

Would that work for your group?  Would you use both of the new CDEs or only one?

**Reply by Ssandor13** [reply]
Just to clarify, this means that 'Single-Cell Genomic" will not be added to either, correct? @jknable

**Comment by GeoffLyle** [comment]
Per @calkinsh discussion on Slack:
Should we also include "Not Reported" and "Other" as permissible values in the new CDEs Source Material and Source Molecule?

**Reply by jknable** [reply]
we have added "Not Reported" as a permissible value.

**Comment by jknable** [comment]
@GeoffLyle We have created these 2 new CDEs for your review.   

We did not add ‘Synthetic’ to library_source_molecule as we need to request a new term from our terminology group.  Can you ask advise if it is synthetic DNA or synthetic RNA? 

**library_source_material
CDE 14808227 Sequencing Library Cellular Material Source**

Permissible   Value | VM Long Name | Concept Codes | VM Definition

Bulk nuclei | Bulk   Nucleus Specimen | C178224 | A   biospecimen consisting of multiple nuclei as a pool.
Bulk tissue | Bulk   Tissue Specimen | C178225 | A   biospecimen either derived from a whole tissue specimen or tissue section,   which may consist of heterogeneous cells or tissues.
Single-cells | Single   Cell Specimen | C178226 | A   biospecimen that contains the contents of a single cell.
Single-nuclei | Single   Nucleus Specimen | C178227 | A   biospecimen that contains the contents of a single nucleus.
Bulk cells | Bulk   Cell Specimen | C178223 | A   biospecimen consisting of multiple cells as a pool.

**library_source_molecule
CDE 14808232 Sequencing Library Molecule Source**

Permissible Value | VM Long Name | Concept Codes | VM Definition

Genomic DNA | Genomic DNA | C95940 | The DNA that is part of the normal   chromosomal complement of an organism.
Metagenomic | Metagenomic | C201925 | Derived from or associated with   material collected from an environmental sample.
Metatranscriptomic | Metatranscriptomic | C201926 | Derived from or associated with   transcripts from an environmental sample.
Transcriptomic | Transcriptomics | C153189 | A study of the complete set of RNA   transcripts that are produced by the genome, under specific circumstances or   in a specific cell.
Transcriptomic Single Cell | Single Cell RNA Sequencing | C171152 | A procedure that can determine the   nucleotide sequence for all of the RNA transcripts in an amplified nucleotide   sample that was derived from a single cell.
Viral Small RNA | Viral Small RNA | C101143 | Small non-coding RNA that is encoded   by viral DNA and transcribed by virally-infected cells.

**Reply by dunnpa** [reply]
Sequence Read Archive has 'VIRAL RNA'. WHat are thoughts on updating 'Viral Small RNA' to 'Viral RNA'?

**Reply by dunnpa** [reply]
Sequence Read Archive has 'GENOMIC'. Thoughts on editing 'Genomic DNA' to 'Genomic'?

**Reply by hbeale** [reply]
@dunnpa editing 'Genomic DNA' to 'Genomic' seems potentially confusing. I'm fine with  'Viral Small RNA' to 'Viral RNA', but I would defer to anyone else's opinion; it's not my area of expertise.

**Comment by hbeale** [comment]
@jknable Geoff asked me to take a look at these. 

## library_source_molecule
I suggest removing "Transcriptomic Single Cell" from library_source_molecule since it is defined by a combination of the "Transcriptomic" library_source_molecule and "Single-cells" library_source_material (as I read single-cells, see below)

I take the VM definition of Transcriptomic ("A study of the complete set of RNA transcripts that are produced by the genome, under specific circumstances or in a specific cell") with a grain of salt. No transcript method captures a complete set of RNA transcripts. 
 
I do not see a need to add Synthetic to library source molecule. 

## library_source_material
I'm confused by the definition of Single-cells in library_source_material. Does the VM definition of "Single-cells" as "A biospecimen that contains the contents of a single cell" encompass that idea that the input of the dataset is many biospecimens? Typically  ~thousands of individual cells undergo library prep in such a way that the resulting dataset will contain sequence data from multiple cells distinguished by barcodes that indicate which data came from the same cell. 

The same issue applies to "Single-nuclei". 

We also need to distinguish this from bulk cells and bulk nuclei. The VM definition "A biospecimen consisting of multiple cells as a pool" could equally apply to my expectation of the input for a single-cell experiment. 

ideas for solving it
* Make specimen plural, i.e. Single Cell Specimens and Single Nucleus Specimens
* Modify the "Single" definitions, e.g. from "A biospecimen that contains the contents of a single cell" to "A biospecimen that contains the contents of one of more cells intended to be distinguished from each other"
* Modify the "bulk" definitions, e.g. from " A biospecimen consisting of multiple nuclei as a pool " to "A biospecimen consisting of multiple nuclei as a pool **not** intended to be distinguished from each other"

**Reply by jknable** [reply]
@hbeale, @GeoffLyle, @jknable 

NCIt terms are created in the singular format, not as plurals or as adjectives or based on case or acronym.  We can add an alternate name for The Federation that would use the plural form of the term associated with that same value.

Regarding the definitions, we are reaching out to the terminologists to see if the existing definitions for the single/bulk permissible values can be clarified as you have suggested; “…distinguished from each other” or “…not intended to be distinguished from each other”.  Will let you know if they have any additional questions.

**Reply by jknable** [reply]
@hbeale, @GeoffLyle, @jknable

**library_source_material**
Per comments regarding the current definition for single and bulk PVs, the terminology group create new concepts for the “Single” PVs and revised the definitions for the “Bulk” concepts.   Please review below and advise if these revisions captured your intent for these PVs.

**•	Single-cell**
o	NCI Concept: Cell Suspension for Single-Cell Assay
o	Synonym: Single Cell Suspension
o	Definition: A dilute suspension of cells intended to be further fractionated for assays focused on single-cells.

**•	Single-nuclei** 
o	NCI Concept: Nuclei Suspension for Single-Nuclei Assay
o	Synonym: Single Nucleus Suspension
o	Definition: A dilute suspension comprised of isolated intact cell nuclei intended to be further fractionated for assays focused on single-nuclei.

**•	Bulk Cells:**
o	NCI Concept: Bulk Cell Specimen
o	NCI Code: C178223
o	Current Definition:  A biospecimen consisting of multiple cells as a pool.
o	New Definition: A biospecimen consisting of multiple cells intended to be analyzed as a pool.

**•	Bulk Nuclei:**
o	NCI Concept: Bulk Nucleus Specimen
o	NCI Code: C178224
o	Current Definition:  A biospecimen consisting of multiple nuclei as a pool.
o	New Definition: A biospecimen consisting of multiple nuclei intended to be analyzed as a pool.

**Reply by hbeale** [reply]
Thanks! I think "intended to be analyzed as a pool" is excellent. I think "isolated" or "separated" would be preferable to "fractionated". "Fractionation" is commonly used with a different meaning in cell biology.

**Reply by hbeale** [reply]
Did we decide to drop Bulk Tissue? I'm not sure what distinguishes it from bulk cells (which might also be heterogeneous). If we decide to keep it, we should also include in the definition that is is intended to be analyzed as a pool.

**Reply by calkinsh** [reply]
@jknable could you also provide the info for Bulk Tissue? I'm trying to confirm internally if those work and how we'd use them

**Reply by jknable** [reply]
@calkinsh  - please see CDE 14808227 Sequencing Library Cellular Material Source.  To avoid conflated data, the original list was split into material and molecule. 
<img width="459" alt="Material-Molecule" src="https://github.com/CBIIT/ccdi-federation-api/assets/158315884/4f9cb233-5718-47c5-bb85-88e7483621ad">

**Comment by hbeale** [comment]
A general discussion about CDE definitions is whether we want to try to make them parallel within a given CDE? 

For example, we have the following definitions for single and bulk RNA-Seq:
A dilute suspension of cells intended to be further fractionated for assays focused on single-cells.
A biospecimen consisting of multiple cells intended to be analyzed as a pool.

Does that mean that the dilute suspension of cells isn't a biospecimen? Could the second definition equally be "multiple cells intended to be analyzed as a pool"? 

Looking forward to hearing from someone who is deeper in CDE land than I am!

**Comment by GeoffLyle** [comment]
For both **library_source_material** ([14808227](https://cadsr.cancer.gov/onedata/dmdirect/NIH/NCI/CO/CDEDD?filter=CDEDD.ITEM_ID=14808227%20and%20ver_nr=1)) and **library_source_molecule** ([14808232](https://cadsr.cancer.gov/onedata/dmdirect/NIH/NCI/CO/CDEDD?filter=CDEDD.ITEM_ID=14808232%20and%20ver_nr=1)), Federation members will review permissible values and their definitions and provide feedback by July 1, 2024.  CDEs will be reviewed for adoption at July 12 Data Harmonization meeting.

**Reply by calkinsh** [reply]
After some internal discussion I am reporting back the following discussion points from some internal stakeholders:
- the library_source_material CDE is ok, and to expand further on the conversation from the call we think we would be ok to classify blood as bulk tissue but a blood derivative (like buffy coat) as bulk cells
- library_source_molecule lead to some additional debate.
   - 1. request to add protien as a value
   - 2. we were wondering why viral RNA is called out separately, and also why it is referred to as RNA when in the other values we use transcriptomic? not a deal breaker, just noted as feeling a little odd or maybe inconsistent 
   - 3. disagreement that "genomic DNA" == "genomic", per Jo Lynne: "genomic doesn't necessarily mean DNA, it could also mean epigenome or something else genome-wide"
   - 4. Can multiple values be selected, or what would be the correct way to describe eCLIP-Seq which is an RNA-protein crosslink or ChIP-Seq which is DNA-protein crosslinked?

**Reply by GeoffLyle** [reply]
Thanks for your response!
a. @jknable Can we add "Protein" as a PV?
b. This may be a value from SRA, not that we should be beholden to values from SRA, but if we can see a use for this value we should include it (Treehouse would not use "Viral RNA", only "Transcriptomic")
c. Agreed that "Genomic DNA" is much more informative and specific than just "Genomic".  I believe it was changed to "Genomic" since it matches values in SRA.  I think in this case we should use "Genomic DNA".  @jknable is it possible to change this PV back.
d. Seems like a good use case for allowing multiple values to be selected.

**Reply by dunnpa** [reply]
c. Agreed that "Genomic DNA" is much more informative and specific than just "Genomic". I believe it was changed to "Genomic" since it matches values in SRA. I think in this case we should use "Genomic DNA". @jknable is it possible to change this PV back.

I also agree "Genomic DNA" is the better term vs. "Genomic". CCDI SRA submissions could deal with this via a transformation of the values while producing the SRA manifest.

**Reply by GeoffLyle** [reply]
@hbeale and I were reviewing the PVs for Library Source Molecule.  For the environmental PVs (Viral RNA, Metagenomic, and Metatranscriptomic), would those not also be included in the other PVs 'Transcriptomic' and 'Genomic DNA'?  Also Treehouse does not have any environmental genomic data to share, do the other Federation members have a use for these PVs?  If not we could pare down the PVs to the broader categories:

- Genomic DNA
- Transcriptomic
- Proteomic
- Not Reported

Is this CDE being used elsewhere besides our use in the CCDI Federation?

**Reply by jknable** [reply]
Regarding "Is this CDE being used elsewhere besides our use in the CCDI Federation?", yes it is so there may be PVs needed by other communities included in the CDE.

**Comment by GeoffLyle** [comment]
From the July 26, 2024 meeting:
**library_source_material** ([14808227](https://cadsr.cancer.gov/onedata/dmdirect/NIH/NCI/CO/CDEDD?filter=CDEDD.ITEM_ID=14808227%20and%20ver_nr=1)) has been accepted as ready for implementation.
**library_source_molecule** ([14808232](https://cadsr.cancer.gov/onedata/dmdirect/NIH/NCI/CO/CDEDD?filter=CDEDD.ITEM_ID=14808232%20and%20ver_nr=1)) request to change "Genomic" to "Genomic DNA" and then accept the CDE.  Request made by @calkinsh to add "Protein"/"Proteomic" on August 1st, 2024.  @jknable can we add a "protein"/"proteomic" PV to this CDE?  I believe @calkinsh can provide terms and a definition.

**Reply by jknable** [reply]
@calkinsh Change "Genomic" to "Genomic DNA" has been made, looking into options for "protein"/"proteomic" PV.

**Reply by calkinsh** [reply]
great, thank you! Let me know if I should provide suggested values for protein/proteomic, or if there is an existing term that can be reused for the CDE

**Reply by jknable** [reply]
@calkinsh if you have some examples of how you would use this value, that would be helpful.  

Since the intent of this CDE is to describes the types of molecules studied, we would recommend "protein" rather than "proteomics", but please advise if this would not work for your use case.  Paired components in a single permissible value like "protein"/"proteomic" are not recommended. Here are the NCIt definitions for both:

•	Protein (C17021) - A group of complex organic macromolecules composed of one or more chains (linear polymers) of alpha-L-amino acids linked by peptide bonds and ranging in size from a few thousand to over 1 million Daltons. Proteins are fundamental genetically encoded components of living cells with specific structures and functions dictated by amino acid sequence.

•	Proteomics (C20085) - The global analysis of cellular proteins. Proteomics uses a combination of sophisticated techniques including two-dimensional (2D) gel electrophoresis, image analysis, mass spectrometry, amino acid sequencing, and bio-informatics to resolve comprehensively, to quantify, and to characterize proteins. The application of proteomics provides major opportunities to elucidate disease mechanisms and to identify new diagnostic markers and therapeutic targets.

**Reply by calkinsh** [reply]
@jknable I'm stuck on this because based on these descriptions, protein is more accurate - the CDE is intended to describe what is being used in the assay, and in that case the descriptor for protein is more accurate than the descriptor for proteomics, which describes a general class of assays. But, the other PVs are generally formatted as descriptors - transcriptomic, metagenomic, etc.  In that context, I feel like a value of "Proteomic" but with a definition more aligned with Protein (C17021) would be the best fit. However, I'm not sure if this is achievable? Would be interested to hear other thoughts.

**Comment by GeoffLyle** [comment]
From Meeting on August 9th, 2024:
**Library Source Molecule** [14808232](https://cadsr.cancer.gov/onedata/dmdirect/NIH/NCI/CO/CDEDD?filter=CDEDD.ITEM_ID=14808232%20and%20ver_nr=1) would not add a 'protein' permissible value as it is for nucleic acid data (Definition: The molecular source of the nucleic acid fragments analyzed by high-throughput sequencing.).  The Federation agreed this is understandable, however, it leaves us with an issue of where to document proteomic data.  Another potential CDE option is **Molecular Analysis Analyte Type** [6142394](https://cadsr.cancer.gov/onedata/dmdirect/NIH/NCI/CO/CDEDD?filter=CDEDD.ITEM_ID=6142394%20and%20ver_nr=2).  Federation could potentially adopt both CDEs, however, there are concerns about duplication of data information causing increased complication of use and confusion to end users.  

Federation members should review **Molecular Analysis Analyte Type** [6142394](https://cadsr.cancer.gov/onedata/dmdirect/NIH/NCI/CO/CDEDD?filter=CDEDD.ITEM_ID=6142394%20and%20ver_nr=2) and determine how it fits their data, and how it fits into the currently accepted CDEs.

GL Edit: Aug 12, 2024 - Fix typo - **Library Source Material** -> **Library Source Molecule**

**Comment by GeoffLyle** [comment]
From September 20th meeting:
**Molecular Analysis Analyte Type** [6142394](https://cadsr.cancer.gov/onedata/dmdirect/NIH/NCI/CO/CDEDD?filter=CDEDD.ITEM_ID=6142394%20and%20ver_nr=2) has been deprecated.  caDSR team recommends using **Specimen Molecular Analyte Type** [15063661](
https://cadsr.cancer.gov/onedata/dmdirect/NIH/NCI/CO/CDEDD?filter=CDEDD.ITEM_ID=15063661%20and%20ver_nr=1)  
Definition: "The sample or material being subjected to analysis."

Data Harmonization team reviewed the **Specimen Molecular Analyte Type** CDE and Permissible Values.  All current Federation members preferred using the CDE's Permissible Values over **Library Source Molecule** [14808232](https://cadsr.cancer.gov/onedata/dmdirect/NIH/NCI/CO/CDEDD?filter=CDEDD.ITEM_ID=14808232%20and%20ver_nr=1).  Concern over ambiguities in Permissible Values 'mRNA' and 'Total RNA' being similar to other RNA-Seq library construction terms 'Poly-A Enriched Genomic Library' and 'rRNA Depletion'.  Potential work-arounds were discussed: 

- Use only the Permissible Value 'RNA' for this CDE (15063661) and allow users to determine the library selection method from CDE [6347742 v2](https://cadsr.cancer.gov/onedata/dmdirect/NIH/NCI/CO/CDEDD?filter=CDEDD.ITEM_ID=6347743%20and%20ver_nr=2).  This was an agreed upon solution during the meeting.
- Subset the Permissible Values (similar to Library Source Material #114) and including 'RNA' and removing 'mRNA' and 'Total RNA'.  Meeting attendees agreed with this solution, but noted they would have to discuss this option with other members to ensure full concordance.

TODO:

1. Fill out which Permissible Values from **Specimen Molecular Analyte Type** [15063661](
https://cadsr.cancer.gov/onedata/dmdirect/NIH/NCI/CO/CDEDD?filter=CDEDD.ITEM_ID=15063661%20and%20ver_nr=1) would currently be used by Federation.  These values are being tracked in this [Google Sheet](https://docs.google.com/spreadsheets/d/1eT2ZahJTUq0ZihhVouSSqKQCYO0z2KrFDbgNB06DC2w/edit?gid=2012503625#gid=2012503625).
2. Determine whether we will use the full list of Permissible Values or the subset of values currently being used.
3. Implement the CDE in the API.

**Comment by GeoffLyle** [comment]
From October 04, 2024 meeting:
We approved **Specimen Molecular Analyte Type** [15063661](https://cadsr.cancer.gov/onedata/dmdirect/NIH/NCI/CO/CDEDD?filter=CDEDD.ITEM_ID=15063661%20and%20ver_nr=1) for implementation.  We will subset the permissible values to the following:

- DNA
- Protein
- RNA

Note: Values 'Not Reported' and 'Unknown' were not included as this information should be known when sharing this data.  If a member does not fill out this field, the API should return a _null_ response.  We believe not including these values will make it easier for users to parse this field.
